### PR TITLE
refactor: Add option to convert `Parse.Object` to instance in Cloud Function payload

### PR DESCRIPTION
**Pull request by mtrezza**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/parse-server/issues/8634

### CODE REVIEW

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.3.0-alpha.3](https://github.com/parse-community/parse-server/releases/tag/6.3.0-alpha.3)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.3.0](https://github.com/parse-community/parse-server/releases/tag/6.3.0)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.4.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.4.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.4.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/6.4.0-alpha.1)

File: src/Routers/FunctionsRouter.js [hunk 1]
```diff
@@ -18,21 +18,21 @@ function parseObject(obj) {
     return Object.assign(new Date(obj.iso), obj);
   } else if (obj && obj.__type == 'File') {
     return Parse.File.fromJSON(obj);
-  } else if (obj && obj.__type == 'Pointer') {
+  } else if (obj && obj.__type == 'Pointer' && config.encodeParseObjectInCloudFunction) {
```
Comment (sadortun) [inline src/Routers/FunctionsRouter.js:21]: @mtrezza 

Not sure if i'm missing something, but i get this error

```
TypeError: Cannot read properties of undefined (reading 'encodeParseObjectInCloudFunction')
    at parseObject (/var/www/asdf/node_modules/parse-server/src/Routers/FunctionsRouter.js:21:55)
```

Haven't investigated yet.

File: src/Routers/FunctionsRouter.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { jobStatusHandler } from '../StatusHandler';
 import _ from 'lodash';
 import { logger } from '../logger';
 
-function parseObject(obj) {
+function parseObject(obj, config) {
   if (Array.isArray(obj)) {
     return obj.map(item => {
       return parseObject(item);
```
Comment (sadortun) [inline src/Routers/FunctionsRouter.js:15]: Are you missing `parseObject(obj, config)` here ?

Comment (sadortun) [reply src/Routers/FunctionsRouter.js:15]: NVM, just found issue https://github.com/parse-community/parse-server/pull/8784

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.4.0](https://github.com/parse-community/parse-server/releases/tag/6.4.0)

**Pull request merged**